File: viewpager/src/main/java/com/example/dynamicviewpager/MainActivity.java
package com.example.dynamicviewpager;

import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.tabs.TabLayout;

import java.util.ArrayList;
import java.util.Timer;
import java.util.TimerTask;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        productAdvertise();
        setTimer();
    }

    private void productAdvertise() {

        final int[] currentPage = {0};
        final ArrayList<String> list = new ArrayList<>();

        list.add("https://homepages.cae.wisc.edu/~ece533/images/airplane.png");
        list.add("https://homepages.cae.wisc.edu/~ece533/images/arctichare.png");
        list.add("https://homepages.cae.wisc.edu/~ece533/images/baboon.png");
        list.add("https://homepages.cae.wisc.edu/~ece533/images/barbara.bmp");
        list.add("https://homepages.cae.wisc.edu/~ece533/images/boat.png");

        MainPagerAdapter pagerAdapter = new MainPagerAdapter(getApplicationContext(), list);
        final ViewPagerCustomDuration pager = findViewById(R.id.view_pager);
        pager.setScrollDurationFactor(5);
        pager.setAdapter(pagerAdapter);

        for (int i = 0; i < list.size(); i++) {
            pagerAdapter.addView(getLayoutInflater().inflate(R.layout.item_advertise, null), i);
        }

        pagerAdapter.notifyDataSetChanged();

        TabLayout tabLayout = findViewById(R.id.tabDots);
        tabLayout.setupWithViewPager(pager, true);

        final Handler handler = new Handler();
        final Runnable Update = new Runnable() {
            public void run() {
                if (currentPage[0] == list.size()) {
                    currentPage[0] = 0;
                }
                pager.setCurrentItem(currentPage[0]++, true);
            }
        };
        Timer timer = new Timer(); // This will create a new Thread
        timer.schedule(new TimerTask() { // task to be scheduled
            @Override
            public void run() {
                handler.post(Update);
            }
        }, 500, 1800);
    }

    private void setTimer() {

    }
}

File: CustomViewPager/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.os.Bundle
import android.os.Handler
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.PagerAdapter
import com.example.myapplication.customviewpager.ViewPagerCustomDuration
import com.google.android.material.tabs.TabLayout
import java.util.*


class MainActivity : AppCompatActivity() {

    private val mTag = "MainActivity"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        productAdvertise()

    }

    private fun productAdvertise() {

        val adapter: PagerAdapter = object : FragmentPagerAdapter(supportFragmentManager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

            var colours = getFragmentList()

            override fun getCount(): Int {
                Log.d(mTag, "size = ${colours.size}")
                return colours.size
            }

            override fun getItem(position: Int): Fragment {
                val fragment: Fragment = BannerFragment()
                val args = Bundle()
                args.putString("name", colours[position])
                //args.putInt("identifier", position)
                fragment.arguments = args
                return fragment
            }
        }

        val currentPage = intArrayOf(0)

        val pager = findViewById<ViewPagerCustomDuration>(R.id.view_pager)
        pager.setScrollDurationFactor(5.0)
        pager.adapter = adapter

        val tabLayout: TabLayout = findViewById<TabLayout>(R.id.tabDots)
        tabLayout.setupWithViewPager(pager, true)

        val updateHandler = Handler()
        val updateRunnable = Runnable {
            if (currentPage[0] == getFragmentList().size) {
                currentPage[0] = 0
            }
            pager.setCurrentItem(currentPage[0]++, true)
        }
        val timer = Timer() // This will create a new Thread
        timer.schedule(object : TimerTask() {
            // task to be scheduled
            override fun run() {
                updateHandler.post(updateRunnable)
            }
        }, 500, 1800)
    }

    private fun getFragmentList(): ArrayList<String> {
        val list = ArrayList<String>()
        list.add("https://homepages.cae.wisc.edu/~ece533/images/airplane.png")
        list.add("https://homepages.cae.wisc.edu/~ece533/images/arctichare.png")
        list.add("https://homepages.cae.wisc.edu/~ece533/images/baboon.png")
        list.add("https://homepages.cae.wisc.edu/~ece533/images/barbara.bmp")
        list.add("https://homepages.cae.wisc.edu/~ece533/images/boat.png")
        return list
    }

}